src/WebJobs.Script.WebHost/Routing/ScriptRouteHandler.cs

```diff
@@ -0,0 +1,128 @@
+using Microsoft.Azure.WebJobs.Script.WebHost.Security.Authorization;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Http
+{
+    public class ScriptRouteHandler : IWebJobsRouteHandler
+    {
+        private readonly Func<ScriptHost> _scriptHostProvider;
+        private readonly ILoggerFactory _loggerFactory;
+
+        public ScriptRouteHandler(ILoggerFactory loggerFactory, Func<ScriptHost> scriptHostProvider)
+        {
+            _scriptHostProvider = scriptHostProvider;
+            _loggerFactory = loggerFactory;
+        }
+
+        public ScriptRouteHandler(ILoggerFactory loggerFactory, ScriptHost scriptHost)
+        {
+            _loggerFactory = loggerFactory;
+            _scriptHostProvider = () => scriptHost;
+        }
+
+        public async Task InvokeAsync(HttpContext context, string functionName)
+        {
+            IActionResult result = await GetResultAsync(context, functionName);
+
+            var actionContext = new ActionContext
+            {
+                HttpContext = context
+            };
+
+            await result.ExecuteResultAsync(actionContext);
+        }
+
+        private async Task<IActionResult> GetResultAsync(HttpContext context, string functionName)
+        {
+            ScriptHost host = _scriptHostProvider();
+
+            // TODO: FACAVAL This should be improved....
+            FunctionDescriptor descriptor = host.Functions.FirstOrDefault(f => string.Equals(f.Name, functionName));
+
+            if (descriptor == null)
+            {
+                return new NotFoundResult();
+            }
+
+            var routingFeature = context.Features.Get<IRoutingFeature>();
+
+            // Add rounte data to request info
+            // TODO: Keeping this here for now as other code depend on this property, but this can be done in the HTTP binding.
+            context.Items.Add(HttpExtensionConstants.AzureWebJobsHttpRouteDataKey, new Dictionary<string,object>(routingFeature.RouteData.Values));
+
+            context.Features.Set<IFunctionExecutionFeature>(new FunctionExecutionFeature { Descriptor = descriptor });
+
+            bool authorized = await AuthenticateAndAuthorizeAsync(context, descriptor);
+
+            if (!authorized)
+            {
+                return new UnauthorizedResult();
+            }
+
+            Dictionary<string, object> arguments = GetFunctionArguments(descriptor, context.Request);
+
+            // Add the request to the logging scope. This allows the App Insights logger to
+            // record details about the request.
+            ILogger logger = _loggerFactory.CreateLogger(LogCategories.Function);
+            var scopeState = new Dictionary<string, object>()
+            {
+                [ScriptConstants.LoggerHttpRequest] = context.Request
+            };
+
+            using (logger.BeginScope(scopeState))
```

Comment:
Where is all the HttpRequestManager throttling infrastructure that used to be in FunctionsController? It looks like those types were totally deleted? We need those.

Reply:
That has not been plugged in (I'm tracking this [here](https://github.com/Azure/azure-webjobs-sdk-script/blob/642d3b186d0a017783c133240323f8703483e79e/src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs#L45-L55) and have issue #1953 to track the work.

The type still exists in the HTTP binding extension, it just needs to be plugged in.